src/rpc/blockchain.cpp

```
@@ -1754,10 +1754,48 @@ try{
+            "2. \"basepubkeys\"     (array, required) A json array of ordered hex encoded of the compressed base (un-tweaked) public keys that were used in the multisig\n\n"
+            "    [\n"
+            "      \"basepubkey\", (string, required) Hex encoded of the compressed base (un-tweaked) public key that was used in the multisig\n\n"
+            "      ,...\n"
+            "    ]\n"
+            "3. \"nmultisig\"     (numeric, required) Number of required signatures for a multisig transaction (n of M)\n"
+            "4. \"kycaddress\"    (string, optional) Base58 KYC address\n"
+            "\nExamples:\n"
+            + HelpExampleCli("addmultitowhitelist", "\"1dncVuBznaXPDNv8YXCKmpfvoDPNZ288MhB \"[\\\"038f9c608ded55e89aef8ade69b90612510dbd333c8d63cbe1072de9049731bb58\\\",\\\"028f9c608ded55e89aef8ade69b90612510dbd333c8d63cbe1072de9049731bb58\\\"]\" 1, \"2dncVuBznaXPDNv8YXCKmpfvoDPNZ288MhB\"")
+            + HelpExampleRpc("addmultitowhitelist", "\"1dncVuBznaXPDNv8YXCKmpfvoDPNZ288MhB \"[\\\"038f9c608ded55e89aef8ade69b90612510dbd333c8d63cbe1072de9049731bb58\\\",\\\"028f9c608ded55e89aef8ade69b90612510dbd333c8d63cbe1072de9049731bb58\\\"]\" 1, \"2dncVuBznaXPDNv8YXCKmpfvoDPNZ288MhB\"")
+                        );
+#ifdef ENABLE_WALLET
+    if (request.params[1].isNull())
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, argument 2 must be non-null");
+    try{
+        if(nparams == 3){
+            addressWhitelist.add_multisig_whitelist(request.params[0].get_str(), request.params[1].get_array(), request.params[2].get_int());
+        } else {
+            addressWhitelist.add_multisig_whitelist(request.params[0].get_str(), request.params[1].get_array(),
+            request.params[3].get_str(), request.params[2].get_int());
+        }
+    } catch(std::invalid_argument e){
+         throw JSONRPCError(RPC_INVALID_PARAMETER, e.what());
+    }
+#endif //#ifdef ENABLE_WALLET
+
+    return NullUniValue;
```

This should probably return an error.